### PR TITLE
chore: upgrade libs wagmi, rainbowkit, viem

### PR DESCRIPTION
**Pull request by 0xtiti**
## Description:
- New API that includes Wagmi's createConfig and replaces getDefaultWallets and connectorsForWallets
- Instantiating wallet connectors and passing projectId and chains is no longer required.

## Checklist before requesting a review

- [x] I have conducted a self-review of my code.
- [ ] If it is a core feature, I have included comprehensive tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one sentence about this update.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,12 +24,13 @@
     "src/**/*.{js,jsx,ts,tsx,css,scss,md,html}": "prettier --write --ignore-unknown"
   },
   "dependencies": {
-    "@rainbow-me/rainbowkit": "1.0.9",
+    "@rainbow-me/rainbowkit": "2.0.2",
+    "@tanstack/react-query": "^5.28.0",
```
Comment (turtlemoji) [inline package.json]: fix package (remove caret)

**Comment by turtlemoji** [comment]
check semantic pr name

File: package.json (hunk 1 above)
Comment (0xtiti) [reply package.json]: fixed

File: src/main.tsx [hunk 2]
```diff
@@ -2,62 +2,64 @@ import React from 'react';
...
-import { injectedWallet, rainbowWallet, walletConnectWallet } from '@rainbow-me/rainbowkit/wallets';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { createConfig } from 'wagmi';
 import { connectorsForWallets } from '@rainbow-me/rainbowkit';
-
-import { configureChains, createConfig, WagmiConfig } from 'wagmi';
-import { alchemyProvider } from 'wagmi/providers/alchemy';
-import { publicProvider } from 'wagmi/providers/public';
+import { rainbowWallet, walletConnectWallet, injectedWallet } from '@rainbow-me/rainbowkit/wallets';
+import { WagmiProvider, http } from 'wagmi';
```
Comment (0xArdy) [inline src/main.tsx]: i think you can group these imports
```suggestion
import { createConfig, WagmiProvider, http } from 'wagmi';
import { connectorsForWallets } from '@rainbow-me/rainbowkit';
import { rainbowWallet, walletConnectWallet, injectedWallet } from '@rainbow-me/rainbowkit/wallets';
```

**Review by 0xArdy — DISMISSED** [review]
great work! :rocket: 
now after resolving the conflicts you can Squash and merge

**Review by 0xArdy — APPROVED** [review]

**Pull request merged**